src/Storages/S3Queue/S3QueueTableMetadata.cpp

```diff
@@ -0,0 +1,115 @@
+#include <config.h>
+
+#if USE_AWS_S3
+
+#    include <IO/Operators.h>
+#    include <IO/ReadBufferFromString.h>
+#    include <Storages/S3Queue/S3QueueSettings.h>
+#    include <Storages/S3Queue/S3QueueTableMetadata.h>
+#    include <Storages/StorageS3.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int METADATA_MISMATCH;
+}
+
+S3QueueTableMetadata::S3QueueTableMetadata(const StorageS3::Configuration & configuration, const S3QueueSettings & engine_settings)
+{
+    format_name = configuration.format;
+    after_processing = engine_settings.after_processing.toString();
+    mode = engine_settings.mode.toString();
+    s3queue_max_set_size = engine_settings.s3queue_max_set_size;
+    s3queue_max_set_age_s = engine_settings.s3queue_max_set_age_s;
+}
+
+void S3QueueTableMetadata::write(WriteBuffer & out) const
+{
+    out << "metadata format version: 1\n"
```

Comment:
may be it could be more convenient to use json format?
similar to
https://github.com/ClickHouse/ClickHouse/blob/cc3897a84aa607d172f593539c22153f200584e1/src/Interpreters/ClusterDiscovery.cpp#L526-L533